api/cpp/esp-idf/slint/README.md

```diff
@@ -11,80 +11,139 @@ It has been tested on ESP32-S3 devices.
-
+6. Remove `main/slint-hello-world.c`.
+7. Create a new file `main/slint-hello-world.cpp` with the following contents:
+```cpp
+#include <stdio.h>
+#include <esp_err.h>
+#include <bsp/display.h>
+#include <bsp/esp-bsp.h>
+#include <slint-esp.h>
+
+#if defined(BSP_LCD_DRAW_BUFF_SIZE)
+#    define DRAW_BUF_SIZE BSP_LCD_DRAW_BUFF_SIZE
+#else
+#    define DRAW_BUF_SIZE (BSP_LCD_H_RES * CONFIG_BSP_LCD_DRAW_BUF_HEIGHT)
+#endif
+
+#include "appwindow.h"
+
+extern "C" void app_main(void)
+{
+    /* Initialize display  */
+    esp_lcd_panel_io_handle_t io_handle = NULL;
+    esp_lcd_panel_handle_t panel_handle = NULL;
+    const bsp_display_config_t bsp_disp_cfg = {
+        .max_transfer_sz = DRAW_BUF_SIZE * sizeof(uint16_t),
+    };
+    bsp_display_new(&bsp_disp_cfg, &panel_handle, &io_handle);
+
+     /* Set display brightness to 100% */
+    bsp_display_backlight_on();
+
+    std::optional<esp_lcd_touch_handle_t> touch_handle;
+
+    /* Allocate a drawing buffer */
+    static std::vector<slint::platform::Rgb565Pixel> buffer(BSP_LCD_H_RES * BSP_LCD_V_RES);
+
+    /* Initialize Slint's ESP platform support*/
+    slint_esp_init(slint::PhysicalSize({ BSP_LCD_H_RES, BSP_LCD_V_RES }), panel_handle,
+                                       touch_handle, buffer);
+    /* Instantiate the UI */
+    auto ui = AppWindow::create();
+    /* Show it on the screen and run the event loop */
+    ui->run();
+}
 ```
-error: the `-Z` flag is only accepted on the nightly channel of Cargo, but this is the `stable` channel
+8. Create `main/appwindow.slint` with the following contents:
 ```
+import { VerticalBox, AboutSlint } from "std-widgets.slint";
+export component AppWindow {
+    VerticalBox {
+        AboutSlint {}
+        Text {
+            text: "Hello World";
+            font-size: 18px;
+            horizontal-alignment: center;
+        }
+    }
+}
+```
+9. Edit `main/CMakeLists.txt` to adjust for the new `slint-hello-world.cpp`, add `slint` as required component,
+   and instruction the build system to compile `appwindow.slint` to `appwindow.h`. The file should look like this:
+```cmake
+idf_component_register(SRCS "slint-hello-world.cpp" INCLUDE_DIRS "." REQUIRES slint)
+slint_target_sources(${COMPONENT_LIB} appwindow.slint)
+```
+10. Open the configuration editor with `idf.py menuconfig`:
+    1. Change the stack size under `Component config --> ESP System Settings --> Main task stack size` to at least `8192`. You may need to tweak this value in the future if you run into stack overflows.
+    2. Add support for C++ exceptions under `Compiler Options -> Enable C++ exceptions`.
+    3. You may need additional device-specific settings. For example if your device has external SPI RAM,
+       you may need to enable that. For details for ESP32-S3 based devices see how to [Configure the PSRAM](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-guides/flash_psram_config.html?highlight=psram#configure-the-psram).
```

Comment:
Is the highlight in the URL on purpose?

Reply:
Oops, no - I just copy and pasted the anchor. Will fix.